mitiq/benchmarks/tests/test_qpe_circuits.py

```diff
@@ -0,0 +1,204 @@
+                cirq.CNOT(qubits[1], qubits[3]),
+            ),
+            cirq.Moment(
+                cirq.CNOT(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                cirq.CNOT(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                cirq.CNOT(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                cirq.CNOT(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                cirq.SWAP(qubits[0], qubits[2]),
+            ),
+            cirq.Moment(
+                cirq.H(qubits[0]),
+                cirq.H(qubits[1]),
+                cirq.H(qubits[2]),
+            ),
+            cirq.Moment(
+                (cirq.CNOT**-1.0).on(qubits[0], qubits[1]),
+            ),
+            cirq.Moment(
+                (cirq.CNOT**-1.0).on(qubits[0], qubits[2]),
+            ),
+            cirq.Moment(
+                (cirq.CNOT**-1.0).on(qubits[1], qubits[2]),
+            ),
+        ]
+    )
+    assert _equal(generated_circuit, expected_circuit)
+
+
+def test_circuit_QPE_TGate():
+    generated_circuit = generate_qpe_circuit(3, cirq.T)
+    qubits = cirq.LineQubit.range(3 + 1)
+    expected_circuit = cirq.Circuit(
+        [
+            cirq.Moment(
+                cirq.H(qubits[0]),
+                cirq.H(qubits[1]),
+                cirq.H(qubits[2]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[2], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[1], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[1], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**0.25).on(qubits[0], qubits[3]),
+            ),
+            cirq.Moment(
+                cirq.SWAP(qubits[0], qubits[2]),
+            ),
+            cirq.Moment(
+                cirq.H(qubits[0]),
+                cirq.H(qubits[1]),
+                cirq.H(qubits[2]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**-0.25).on(qubits[0], qubits[1]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**-0.25).on(qubits[0], qubits[2]),
+            ),
+            cirq.Moment(
+                (cirq.CZ**-0.25).on(qubits[1], qubits[2]),
+            ),
+        ]
+    )
+    assert _equal(generated_circuit, expected_circuit)
+
+
+def test_phase_angle_estimation():
+    def _binary_to_int(input_array):
+        result_string = ""
+        for i in range(len(input_array)):
+            result_string += str(input_array[i])
+            result = int(result_string, 2)
+        return result
```

Comment:
```suggestion
        return int(result_string, 2)
```
This should work as well